### PR TITLE
[Python] Clean up python client code, add logging.

### PR DESCRIPTION
**Pull request by kacpermuda**
### Problem

After debugging some user issues i feel like we could add some more logs, so that we can deduce more information about what's happening inside OpenLineage.

Also, at the moment the whole process of resolving what transport we will have is distributed in three places (factory, client and utils), i moved everything to one location so it's easier to grasp.

### One-line summary:

Clean up python client code, refactor logging in all python modules.

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Comment by kacpermuda** [comment]
Probably #2655 should be merged first, to have the CI fix in a separate small PR

File: client/python/openlineage/client/client.py [hunk 1]
```diff
@@ -127,8 +105,150 @@ def filter_event(
...
 
+    def emit(self, event: Event) -> None:
+        from typing import get_args
+
+        if type(event) not in get_args(Event):
+            msg = "`emit` only accepts RunEvent, DatasetEvent, JobEvent classes"
+            raise ValueError(msg)
+        if not self.transport:
+            log.error("Tried to emit OpenLineage event, but transport is not configured.")
+            return
```
Comment (dolfinus) [inline client/python/openlineage/client/client.py:116]: Probably it's better to raise exception in OpenLineageClient constructor if transport is not provided

Comment (kacpermuda) [reply client/python/openlineage/client/client.py:116]: In OpenLineageClient constructor we are using `console` Transport when there is no transport provided and we emit a warning. What you propose would require changing that behaviour, am i correct? IMHO, we should not remove this `console` fallback but maybe there are some arguments for it?

Comment (dolfinus) [reply client/python/openlineage/client/client.py:116]: If no transport is explicitly set, leading to fallback to `ConsoleTransport`, is `not self.transport` case is even possible?

Comment (kacpermuda) [reply client/python/openlineage/client/client.py:116]: Not unless explicitly remove or something breaks. I think it's more of a safety measure here.

**Pull request merged**